### PR TITLE
Add spike devices

### PR DESCRIPTION
**Pull request by joey0320**
**Related PRs / Issues**:

**Type of change**:
- [ ] Bug fix
- [ ] New feature
- [ ] Other enhancement

**Impact**:
- [ ] RTL change
- [ ] Software change (RISC-V software)
- [ ] Build system change
- [ ] Other

**Contributor Checklist**:
- [ ] Did you set `main` as the base branch?
- [ ] Is this PR's title suitable for inclusion in the changelog and have you added a `changelog:<topic>` label?
- [ ] Did you state the type-of-change/impact?
- [ ] Did you delete any extraneous prints/debugging code?
- [ ] Did you mark the PR with a `changelog:` label?
- [ ] (If applicable) Did you update the conda `.conda-lock.yml` file if you updated the conda requirements file?
- [ ] (If applicable) Did you add documentation for the feature?
- [ ] (If applicable) Did you add a test demonstrating the PR?

- [ ] (If applicable) Did you mark the PR as `Please Backport`?

### CODE REVIEW

File: docs/Software/Spike.rst [hunk 1]
```diff
@@ -50,3 +50,24 @@ Spike-as-a-Tile can be configured with custom IPC, commit logging, and other beh
...
+Example devices are in the ``toolchains/riscv-tools/riscv-spike-devices`` directory.
+These devices are compiled as a shared library that can be dynamically linked to Spike.
+
+To compile these plugins, run ``make`` inside ``toolchains/riscv-tools/riscv-spike-devices``. This will generate a ``libspikedevices.so``.
+
+To hook up a block device to spike and provide a default image to initialize the block device, run
+
+.. code-block:: shell
+   spike --extlib=<path to libspikedevices.so> --device="blockdev,img=<path to Linux image>" <path to kernel binary>
+
```
Comment (jerryz123) [inline docs/Software/Spike.rst:68]: I thought if we install libspikedevices to $RISCV, we can just do `--extlib=libspikedevices.so`

**Review by jerryz123 — CHANGES_REQUESTED** [review]
Can you build and install this library at the end of `build-toolchain-extra.sh`?  `spike-devices` should install to the `$RISCV` directory

File: docs/Software/Spike.rst (hunk 1 above)
Comment (joey0320) [reply docs/Software/Spike.rst:68]: good idea

Comment (joey0320) [reply docs/Software/Spike.rst:68]: done

File: scripts/build-toolchain-extra.sh [hunk 2]
```diff
@@ -125,4 +125,11 @@ cd generators/testchipip/uart_tsi
 make
 cp uart_tsi $RISCV/bin
 
+echo '==>  Installing spike-devices'
+cd $RDIR
+git submodule update --init toolchains/riscv-tools/riscv-spike-devices
+cd toolchains/riscv-tools/riscv-spike-devices
+make
+cp libspikedevices.so $RISCV/lib
```
Comment (jerryz123) [inline scripts/build-toolchain-extra.sh:133]: You can just do `make install PREFIX=$RISCV`

File: docs/Software/Spike.rst [hunk 3]
```diff
@@ -50,3 +50,24 @@ Spike-as-a-Tile can be configured with custom IPC, commit logging, and other beh
...
+However, you may want to add custom device models into Spike such as a block device.
+Example devices are in the ``toolchains/riscv-tools/riscv-spike-devices`` directory.
+These devices are compiled as a shared library that can be dynamically linked to Spike.
+
+To compile these plugins, run ``make`` inside ``toolchains/riscv-tools/riscv-spike-devices``. This will generate a ``libspikedevices.so``.
+
+To hook up a block device to spike and provide a default image to initialize the block device, run
+
+.. code-block:: shell
+   spike --extlib=libspikedevices.so --device="blockdev,img=<path to Linux image>" <path to kernel binary>
```
Comment (jerryz123) [inline docs/Software/Spike.rst:67]: Can we name the `blockdev` devices `iceblk`? I think we should make it clear that this is a block device that matches the `iceblk` peripheral, and requries using the `iceblk` driver.

**Review by abejgonzalez — APPROVED** [review]

**Review by kevindna — COMMENTED** [review]
why did you do this? for clarity?

**Comment by jerryz123** [comment]
I know it's not really a high priority item, but we should at least make some minimal effort to maintain the commit log. 😁 

`commit --amend` exists for a reason.

**Comment by joey0320** [comment]
> I know it's not really a high priority item, but we should at least make some minimal effort to maintain the commit log. 😁
> 
> `commit --amend` exists for a reason.

Yeah... I concede... my commit messages are bad

**Comment by jerryz123** [comment]
I don't think this rebase did the right thing

**Review by jerryz123 — APPROVED** [review]

**Pull request merged**